pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,230 @@
+	locationCredsSecretFlag        = "locationCredsSecret"
+	locationSecretFlag             = "locationSecret"
+	defaultKanisterNamespace       = "kanister"
+	defaultRepositoryServerHost    = "localhost"
+)
+
+type repositoryServerParams struct {
+	tls                             string
+	repositoryServerUser            string
+	repositoryServerUserAccess      string
+	repositoryServerAdminUser       string
+	repositoryServerAdminUserAccess string
+	repositoryPassword              string
+	prefix                          string
+	location                        string
+	locationCreds                   string
+}
+
+func newRepositoryServerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "repository-server",
+		Short: "Create a new RepositoryServer",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createNewRepositoryServer(cmd, args)
+		},
+	}
+
+	cmd.PersistentFlags().StringP(tlsSecretFlag, "t", "", "name of the tls secret needed for secure kopia client and kopia repository server communication")
+	cmd.PersistentFlags().StringP(repoServerUserFlag, "u", "", "name of the user to be created for the kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerUserAccessSecretFlag, "k", "", "name of the secret having access credentials of the users that can connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoAdminUserSecretFlag, "a", "", "name of the secret having admin credentials to connect to connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoPasswordSecretFlag, "r", "", "name of the secret containing password for the kopia repository")
+	cmd.PersistentFlags().StringP(prefixFlag, "p", "", "prefix to be set in kopia repository")
+	cmd.PersistentFlags().StringP(repoServerAdminUserFlag, "z", "", "kopia repository server admin user name")
+	cmd.PersistentFlags().StringP(locationSecretFlag, "l", "", "name of the secret containing kopia repository storage location details")
+	cmd.PersistentFlags().StringP(locationCredsSecretFlag, "c", "", "name of the secret containing kopia repository storage credentials")
+
+	_ = cmd.MarkFlagRequired(tlsSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoAdminUserSecretFlag)
+	_ = cmd.MarkFlagRequired(repoPasswordSecretFlag)
+	_ = cmd.MarkFlagRequired(prefixFlag)
+	_ = cmd.MarkFlagRequired(locationSecretFlag)
+	_ = cmd.MarkFlagRequired(locationCredsSecretFlag)
+	return cmd
+}
+
+func createNewRepositoryServer(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+	if len(args) > 0 {
+		return newArgsLengthError("expected 0 args. Got %#v", args)
+	}
+
+	rsParams, err := generateRepositoryServerParams(cmd)
+	if err != nil {
+		return err
+	}
+
+	repositoryServer, err := validateSecretsAndConstructRepositoryServer(rsParams)
+	if err != nil {
+		return err
+	}
+	_, crCli, _, _ := initializeClients()
```

Comment:
lets not ignore the error coming from initializeClients

Reply:
Sure ! Added Error handling for clients